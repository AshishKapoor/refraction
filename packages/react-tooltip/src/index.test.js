/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import { mount } from 'enzyme';
import { Tooltip, Container } from '.';

jest.mock('@quid/react-popover', () => ({
  __esModule: true,
  Popover: ({ children, ...props }) => children(props),
  Container: 'x-popover-container',
  Arrow: 'x-popover-arrow',
}));

it('renders the Container component', () => {
  const ref = React.createRef();
  const arrowRef = React.createRef();
  const wrapper = mount(
    <Container
      ref={ref}
      arrowProps={{ ref: arrowRef, style: { left: 0, top: 0 } }}
    >
      foobar
    </Container>
  );
  expect(wrapper).toMatchSnapshot();
});

it('forwards props to Popover', () => {
  const wrapper = mount(
    <Tooltip renderTooltip={() => <x-tooltip />}>
      {({ mouseMoveBehavior }: any) => (
        <x-content mouseMoveBehavior={mouseMoveBehavior} />
      )}
    </Tooltip>
  );

  expect(wrapper.find('x-content').prop('mouseMoveBehavior')).toBe(true);
});
