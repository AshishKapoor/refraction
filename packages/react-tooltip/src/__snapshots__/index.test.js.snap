// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Container component 1`] = `
<Container
  arrowProps={
    Object {
      "ref": Object {
        "current": <x-popover-arrow />,
      },
      "style": Object {
        "left": 0,
        "top": 0,
      },
    }
  }
>
  <ForwardRef
    arrowProps={
      Object {
        "ref": Object {
          "current": <x-popover-arrow />,
        },
        "style": Object {
          "left": 0,
          "top": 0,
        },
      }
    }
    className="css-yfonvn-Container emotion-0"
  >
    <x-popover-container
      arrowProps={
        Object {
          "ref": Object {
            "current": <x-popover-arrow />,
          },
          "style": Object {
            "left": 0,
            "top": 0,
          },
        }
      }
      className="css-yfonvn-Container emotion-0"
    >
      foobar
      <x-popover-arrow />
    </x-popover-container>
  </ForwardRef>
</Container>
`;
